docs/Using Price Feeds/historical-price-data.html

```diff
@@ -29,19 +29,93 @@
+![getRoundData Sequence Diagram](/images/data-feed/getRoundData-sequence.png)
 
-The `updatedAt` data feed property is the timestamp of an answered round and `answeredInRound` is the round it was updated in. You can check `answeredInRound` against the current `roundId`. If `answeredInRound` is less than `roundId`, the answer is being carried over. If `answeredInRound` is equal to `roundId`, then the answer is fresh.
+Note that roundIds have different meanings in proxy contracts and in aggregator contracts. Let's clarify this:
+
+### `roundId` in Aggregator (aggregatorRoundId)
+
+Oracles provide periodic data updates to the aggregators. Data feeds are updated in **rounds**. Rounds are identified by their `roundId`, which increases with each new round. This increase may not be monotonic. Knowing the `roundId` of a previous round allows contracts to consume historical price data.
+In the following, we are going to name this roundId as **aggregatorRoundId** so that we differentiate it from the proxy's roundId.
+
+
+### `roundId` in Proxy
+
+Because a proxy has references to the current and all previous underlying aggregators , it needs a way to fetch data from the right aggregator. 
+The `roundId` is computed as the following in the [proxy contract](https://github.com/smartcontractkit/chainlink/blob/develop/contracts/src/v0.7/dev/AggregatorProxy.sol):
+
+```solidity
+roundId = uint80((uint256(phaseId) << 64) | aggregatorRoundId);
+```
+
+where:
+
+- `phaseId` is incremented each time the underlying aggregator implementation has been updated. It is used as key to find the aggregator address. 
+- `aggregatorRoundId` is the aggregator roundId. The id starts at 1.
+- `roundId` computed round id. From the above formula, you can think of it as a large number containing `phaseId` and `aggregatorRoundId`.
+
+> 📘 Remark
+>
+> Note the formula above ensures that no matter how many times the underlying aggregator is changed, the proxy `roundId` will always be increasing.
+>
+
+** Example:**
+
+When querying the historical data, it is important to know when we reach the end of the history of the underlying aggregator. Let's illustrate this with an example:
+At the time of writing, `latestRoundData` [function](/docs/price-feeds-api-reference/#latestrounddata) of  [LINK / USD feed on Ethereum Mainnet](/docs/ethereum-addresses/) returns `roundId = 92233720368547771158`. Using this value, we can compute the `phaseId` and `aggregatorRoundId`:
+
+- `phaseId = 92233720368547771158 >> 64` . Right shifting an integer by 64 bits is equivalent to dividing it by 2^64: `phaseId = 92233720368547771158/ 2^64 = 5` . The current phase id is 5 , which means that this proxy has had 5 underlying aggregators since its initial deployment.
+- `aggregatorRoundId = uint64(92233720368547771158)`. This will retrieve the first 64 bits from the right. If you are calculating this off-chain, you can do the following in javascript:
```

Comment:
```suggestion
- `phaseId = 92233720368547771158 >> 64`: Right shifting an integer by 64 bits is equivalent to dividing it by 2^64: `phaseId = 92233720368547771158/ 2^64 = 5`. The current phase id is 5 , which means that this proxy has had 5 underlying aggregators since its initial deployment.
- `aggregatorRoundId = uint64(92233720368547771158)`: This retrieves the first 64 bits from the right. To calculate this off-chain, you can use the following JavaScript example:
```